### PR TITLE
Add Mermaid chat-card browser QA harness

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary
- Adds a debug-only `/__debug__/mermaid-chat-cards` route that renders assistant, user, disabled, invalid Mermaid, Graphviz, and artifact-style fixtures through the real shared renderer/card components.
- Adds Playwright browser smoke coverage for the harness, including offline auth seeding, readiness/setup gate absence, assistant/artifact Mermaid controls, and source-only fallback paths.
- Fixes shared Markdown Mermaid fence matching so React dev/Strict render replay does not fall back to `CodeBlock`, while preserving global code-block indices.

## Change summary
TODO: human requester must replace this with a human-written explanation of what changed and why these implementation choices were made before merge.

## Test Plan
- [x] `bunx vitest run src/components/Common/__tests__/Markdown.mermaid.test.tsx src/components/Common/__tests__/MermaidDiagramBlock.test.tsx src/components/Common/__tests__/MermaidPreviewDialog.test.tsx src/components/Common/Playground/__tests__/Message.mermaid-rendering.test.tsx src/components/Common/QuickChatHelper/__tests__/QuickChatMessage.mermaid.test.tsx src/components/Sidepanel/Chat/__tests__/ArtifactsPanel.mermaid.test.tsx`
- [x] `npx playwright test e2e/smoke/mermaid-chat-cards.spec.ts --reporter=line`
- [x] `NEXT_PUBLIC_API_URL=http://127.0.0.1:8000 bun run compile`
- [x] Browser-plugin manual QA at `http://127.0.0.1:18002/__debug__/mermaid-chat-cards`
- [x] Bandit touched-scope run via main project venv: zero findings, TS/TSX parser errors expected

## Known Notes
- `bunx vitest run src/routes/__tests__/route-governance.metadata-coverage.test.ts` still has existing metadata/inventory baseline failures unrelated to `/__debug__/mermaid-chat-cards`; the targeted smoke-excluded invariant passed.

---
## Summary by cubic
Adds a debug-only `/__debug__/mermaid-chat-cards` page and a Playwright smoke test to validate Mermaid chat/card rendering in the browser without backend gates. Fixes `Markdown` Mermaid fence matching so React StrictMode replays and whitespace-only indented lines don’t break diagram rendering or block indices.

- **New Features**
  - Added `/__debug__/mermaid-chat-cards` harness using real `Markdown` and `MermaidDiagramBlock` with assistant, user (source-only), disabled, invalid, Graphviz/DOT, and artifact fixtures.
  - Bypassed readiness and first‑run gates for `/__debug__` routes in `_app.tsx` and registered route metadata as `internal_qa_debug` with `smoke: "exclude"`.
  - Added Playwright smoke test covering harness load, gate absence, assistant/artifact controls, and source-only fallbacks.

- **Bug Fixes**
  - Fixed Mermaid fence matching across React StrictMode replays and ignored whitespace‑only indented lines, keeping `blockIndex` stable and preventing `CodeBlock` fallbacks.
  - Added regression tests for StrictMode replay and whitespace-only indent cases.

<sup>Written for commit 8422bc9b12d72959c2b9a70a65fc91017cf9de7d. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_server/pull/2306?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/packages/ui/src/components/Common/Markdown.tsx [hunk 1]
```diff
@@ -38,6 +38,7 @@ const SAFE_URL_PROTOCOL = /^(https?:|mailto:|tel:|blob:)/i
 const DATA_IMAGE_URL_PROTOCOL = /^data:image\//i
 const FENCE_START = /^(\s*)(`{3,}|~{3,})([^\n]*)$/
 const FENCE_CLOSE = /^\s*(`{3,}|~{3,})\s*$/
+const INDENTED_CODE_LINE = /^(?: {4}|\t)/
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Common/Markdown.tsx]: Fixed in 5e84d59623. Added a regression test showing standalone whitespace before a Mermaid fence previously shifted the block index from 0 to 1, then updated `INDENTED_CODE_LINE` to require non-whitespace content when starting an indented code block.

File: apps/packages/ui/src/components/Common/Markdown.tsx [hunk 2]
```diff
@@ -374,18 +396,21 @@ export function Markdown({
...
             ) {
-              closedMermaidFenceCursorRef.current = closedMermaidFenceIndex + 1
               return (
                 <MermaidDiagramBlock
                   artifactContextId={artifactContextId}
-                  blockIndex={blockIndex}
+                  blockIndex={matchedMermaidFence.blockIndex}
                   enableArtifactAction={enableMermaidArtifactActions}
                   source={value}
                 />
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply apps/packages/ui/src/components/Common/Markdown.tsx:416]: Addressed the scanner-drift case in 5e84d59623 by adding a regression test for standalone whitespace before a Mermaid fence and tightening `INDENTED_CODE_LINE` so whitespace-only lines do not start indented code blocks. I intentionally kept `matchedMermaidFence.blockIndex` for Mermaid artifact IDs because the render-callback `blockIndex` is mutable under React dev/Strict render replay, which is the instability this PR fixed.
